backend/src/main/java/ai/giskard/service/JobService.java

```diff
@@ -0,0 +1,59 @@
+package ai.giskard.service;
+
+import ai.giskard.domain.MLWorkerType;
+import ai.giskard.jobs.GiskardJob;
+import ai.giskard.jobs.JobState;
+import ai.giskard.jobs.JobType;
+import ai.giskard.jobs.UndeterminedJob;
+import ai.giskard.web.dto.JobDTO;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+@Service
+public class JobService {
+
+    public static final int JOB_THREADS = 20;
```

Comment:
Should we rely on a thread pool created in `ai.giskard.config.AsyncConfiguration.getTaskScheduler` ? It's already parameterized with application.yml